contracts/tasks/gen-selector-library.ts

```diff
@@ -0,0 +1,73 @@
+import { task } from 'hardhat/config'
+import '@nomiclabs/hardhat-ethers'
+import * as fs from 'fs'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { selectors } from './lib'
+
+task('gen-selector-library', 'Generates a Solidity library with contract selectors for tests').setAction(
+    async (_args, hre: HardhatRuntimeEnvironment) => {
+        await hre.run('compile')
+
+        // ridiculously, this appears to be the only way to get hardhat to compile a specific subtree
+        // we are only updating the in-memory representation of the config, so it won't write this value out to disk
+        // be careful if you compose this task with other tasks in larger scripts!
+        hre.config.paths.sources = './test/mocks'
```

Comment:
@raulk And presumably the rest works out of the box because it works with standard Hardhat folder structure?